client/src/views/workspaces/WorkspaceSharingModal.vue

```diff
@@ -132,37 +191,54 @@ const props = defineProps<{
   informationManager: InformationManager;
 }>();
 
+interface UserRole {
+  role: [UserTuple, WorkspaceRole | null];
```

Comment:
Might as well extract the role. It was done as `[UserTuple, Workspace | null]` before to avoid declaring an interface, but since you're now declaring an interface, you can write
```
interface XX {
  user: UserTuple;
  role: WorkspaceRole | null;
  isSelected: boolean;
}
```